discord/client.py

```diff
@@ -1996,6 +1996,57 @@ def delete_invite(self, invite):
     # Role management
 
     @asyncio.coroutine
+    def move_role(self, server, role, position):
+        """|coro|
+
+        Moves the specified :class:`Role` to the given position in the :class:`Server`.
+
+        This does **not** edit the role ordering in place.
+
+        Parameters
+        -----------
+        server : :class:`Server`
+            The server the role belongs to.
+        role : :class:`Role`
+            The role to edit.
+        position : int
+            The position to insert the role to.
+
+        Raises
+        -------
+        InvalidArgument
+            If position is 0, or role is server.default_role
+        """
+
+        if position == 0:
+            raise InvalidArgument("Cannot move role to position 0")
+
+        if role == server.default_role:
+            raise InvalidArgument("Cannot move default role")
+
+        url = '{0}/{1.id}/roles'.format(endpoints.SERVERS, server)
+
+        roles = []
+
+        # Roles may come in with multiple roles with the same position attribute - fix this here by recalculating.
+        for r in sorted(server.roles, key=lambda x: x.position):
+            if r != role:
+                roles.append(r.id)
+
+        roles.insert(position, role.id)
+
+        payload = [{"id": z[0], "position": z[1]} for z in zip(roles, range(len(roles)))]
```

Comment:
`payload = [{"id": z[0], "position": z[1]} for z in enumerate(roles)]`
built-in func :+1: